R/comparePlot.R

```diff
@@ -96,35 +94,146 @@ comparePlot <- function(compareDF,
+                        symbolSize = c(7, 7, 7, 3),
+                        symbolShape = c("circle", "circle", "circle", "circle"),
+                        symbolColor = c("darkgoldenrod1", "deepskyblue4", "red3", "grey25"),
+                        transparency = 0.5,
                         crosshair = "grey50",
                         referenceLine = "darkgoldenrod1",
                         refLineThickness = 1,
                         legendPosition = "right",
                         footnote,
                         footnoteSize = 3,
-                        footnoteColor = "black"
-) {
-    set.seed(1954)
-    assertthat::assert_that(sum(apply(compareDF, 2, FUN = is.numeric)) >= 2,
+                        footnoteColor = "black") {
+    assertthat::assert_that(!missing(compareDF),
+                            !is.null(compareDF),
+                            "data.frame" %in% class(compareDF),
+                            sum(apply(compareDF, 2, FUN = is.numeric)) >= 2,
                             msg = "Need at least two numeric columns in compareDF.")
-    assertthat::assert_that(plotType %in% c("canvasXpress", "ggplot"),
-                            msg = "Plot type must be either canvasXpress or ggplot.")
-    assertthat::assert_that(length(symbolSize) == 4,
-                            length(symbolShape) == 4,
-                            length(symbolColor) == 4,
-                            length(symbolFill) == 4,
-                            msg = paste0("All specified symbol arguments must be of length 4, ",
-                                         "including symbolSize, symbolShape, symbolColor, and symbolFill."))
+    plotType <- tolower(plotType)
+    assertthat::assert_that(plotType %in% c("canvasxpress", "ggplot"),
+                                       msg = "Plot type must be either canvasXpress or ggplot.")
+    if (any(is.null(pThreshold),
+            !is.numeric(pThreshold),
+            length(pThreshold) != 1)) {
+        warning("pThreshold must be a singular value of class numeric. Assigning default value '0.01'.")
+        pThreshold <- 0.01
+    }
+
+    if (!is.null(title) &&
+        !all(is.character(title), length(title) == 1)) {
+        warning("title must be a singular value of class character. Assigning default value 'NULL'.")
+        title <- NULL
+    }
+
+    if (!is.null(xlab) &&
+        !all(is.character(xlab), length(xlab) == 1)) {
+        warning("xlab must be a singular value of class character. Assigning default value 'NULL'.")
+        xlab <- NULL
+    }
+
+    if (!is.null(ylab) &&
+        !all(is.character(ylab), length(ylab) == 1)) {
+        warning("ylab must be a singular value of class character. Assigning default value 'NULL'.")
+        ylab <- NULL
+    }
+
+    if (any(is.null(symbolColor),
+            !is.character(symbolColor),
+            length(symbolColor)  != 4,
+            .rgbaConversion(symbolColor) == "invalid value")) {
```

Comment:
This is a wrong usage of the function. .rgbaConversion takes only one value . You are passing a vector

Reply:
done